reply.py

```diff
@@ -22,20 +22,33 @@ async def run(self):
 			async for notification in self.pleroma.stream_mentions():
 				await self.process_notification(notification)
 
-	async def process_notification(self, notification):
+	async def process_notification(self, notification, retry_count=0):
 		acct = "@" + notification['account']['acct']  # get the account's @
 		post_id = notification['status']['id']
-		context = await self.pleroma.status_context(post_id)
 
-		# check if we've already been participating in this thread
-		if self.check_thread_length(context):
+		# catch HTTP 500 and backoff on requests
+		retry_count = retry_count + 1
+		try:
+			context = await self.pleroma.status_context(post_id)
+		except pleroma.BadResponse as exc:
+			if retry_count < 3:
+				await anyio.sleep(2**retry_count)
+				await self.process_notification(notification, retry_count)
+			else:
+				# failed too many times in a row, logging
+				print(f"Received HTTP 500 {retry_count} times in a row, aborting reply attempt.")
 			return
 
-		content = self.extract_toot(notification['status']['content'])
-		if content in {'pin', 'unpin'}:
-			await self.process_command(context, notification, content)
 		else:
-			await self.reply(notification)
```

Comment:
since you have a `return` statement in each `except` block, you can dedent the current else block and remove the else clause like

```
try:
    ...
except ...:
    ...
    return

# normal case here
```